File: backend/TaskApplication/service/StatusService.cs
using TaskApplication.entity;
using TaskApplication.repository;

namespace TaskApplication.service
{
    public class StatusService
    {
        private Repository<Status> StatusRepository = new Repository<Status>("statuses");

        public List<Status> GetAllStatuses()
        {
            return StatusRepository.GetAllItems();
        }
    }
}

File: frontend-angular/src/app/service/status/status-service.ts
import { Injectable } from "@angular/core";
import { Status } from "../../entity/status";
import { ApiService } from "../api/api-service";

@Injectable({
    providedIn: "root"
})
export class StatusService {
    private entityName: string;

    constructor(private service: ApiService<Status>) {
        this.entityName = "Status";
    }

    getAllStatuses() {
        return this.service.getAllItems(this.entityName);
    }
}
